app/lib/search/backend.dart

```diff
@@ -239,32 +239,44 @@ PackageDocument createSdkDocument(PubDartdocData lib) {
 
 class SnapshotStorage {
   final String _latestPath = 'snapshot-latest.json.gz';
+  final String _currentPath = 'snapshot/${versions.runtimeVersion}.json.gz';
   final Storage storage;
   final Bucket bucket;
 
   SnapshotStorage(this.storage, this.bucket);
 
   Future<SearchSnapshot> fetch() async {
+    final current = await _fetch(_currentPath);
+    if (current != null) {
+      return current;
+    }
+    return await _fetch(_latestPath);
```

Comment:
Instead of competing to write out snapshot-latest.json.gz, how about doing a directory listing to find the newest snapshot that's still older than versions.runtimeVersion?

Reply:
Planning that in a subsequent PR (with proper cleanup).